libexec/rtld-elf/rtld.c

```diff
@@ -848,13 +857,21 @@ _rtld(Elf_Addr *sp, func_ptr_type *exit_proc, Obj_Entry **objp)
     } else {				/* Main program already loaded. */
 	dbg("processing main program's program header");
 	assert(aux_info[AT_PHDR] != NULL);
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+	phdr = (const Elf_Phdr *) cheri_clearperm(aux_info[AT_PHDR]->a_un.a_ptr, CHERI_PERM_EXECUTIVE);
```

Comment:
For these you can just do things like `phdr = cheri_clearperm(phdr, ...)` etc and just extend the existing code rather than replacing it.